math/cmath/include/algebra/math/algorithms/matrix/inverse/partial_pivot_lud.hpp

```diff
@@ -50,9 +50,10 @@ struct partial_pivot_lud {
     // Calculate inv(A) = inv(U) * inv(L) * P;
     for (size_type j = 0; j < N; j++) {
       for (size_type i = 0; i < N; i++) {
-        element_getter_t()(inv, i, j) = element_getter_t()(P, 0, i) == j
-                                            ? static_cast<scalar_t>(1.0)
-                                            : static_cast<scalar_t>(0.0);
+        element_getter_t()(inv, i, j) =
+            static_cast<size_type>(element_getter_t()(P, 0, i)) == j
```

Comment:
Not so sure about this one. It seemed to be doing a comparison of two doubles before?

Reply:
Hmm. yes. I think comparing the sizes is better